lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1319,8 +1326,9 @@ static bool isExpressionUnableToInline(Operation *op) {
   for (auto user : op->getUsers()) {
     // If the user is in a different block and the op shouldn't be inlined, then
     // we emit this as an out-of-line declaration into its block and the user
-    // can refer to it.
-    if (user->getBlock() != opBlock)
+    // can refer to it. Does not apply to constants, which can always be inlined
+    // into other blocks.
+    if (user->getBlock() != opBlock && !isConstantExpression(op))
```

Comment:
Is this "required" for correctness or are you just trying to clean up the generated IR?

If this is for prettiness, I think we should handle this in rtlcleanup and shouldn't be part of this patch.  If we need it for some reason, I'd love to know why.  Also, in that case, please hoist `isConstantExpresssion(op)` out of the getUsers loop since it is loop invariant

Reply:
Noep this isn't needed for correctness. It was merely here to encourage inlining of constants. But with the rationale that we should rather push duplicates of constants into the blocks where they're used if we want them to be inlined, this change isn't needed :+1: Will drop.